### PR TITLE
Android SDK and gradle updates.

### PR DESCRIPTION
**Pull request by willdenne**
Description: 
The purpose of this PR is to update most of the dependencies, target the latest Android SDK, remove some unused code, and remove jcenter() from the gradle. This is necessary for M1 macs and newer versions of AS.

@cxw320 and myself will be changing around the database and fixing a lot of bugs in the coming days, so this will be the first PR of few more.

### CODE REVIEW

**Review by jollygreenegiant — APPROVED** [review]

**Comment by rowildcard** [comment]
Test verified the app works as expected on the following devices:
One Plus, Android 9
Samsung Galaxy S8, Android 8
emulator Android 12 and 13

**Comment by rhsalas3** [comment]
@willdenne 

While testing we did notice that on an Android 7 (API 24) device we could not install the Google ARCore required for head tracking. The minSDKVersion is set to 24 for the Vocable project, but it looks like [ARCore requires Android 8.1 or newer](https://developers.google.com/ar/devices) (API 27). That may be something to be updated at some other time, we just wanted to make sure to let you all know what we found. I'll add a ticket to the vocable-android board.

The site we referenced for API <-> Android version numbers: https://developer.android.com/studio/releases/platforms#12

File: app/src/main/java/com/willowtree/vocable/presets/CategoriesFragment.kt [hunk 1]
```diff
@@ -56,7 +56,7 @@ class CategoriesFragment : BaseFragment<CategoriesFragmentBinding>() {
         categories?.forEachIndexed { index, category ->
             val categoryButton =
                 CategoryButtonBinding.inflate(inflater, binding.categoryButtonContainer, false)
-            with(categoryButton.root as CategoryButton) {
+            with(categoryButton.root) {
```
Comment (CarolineLaw) [inline app/src/main/java/com/willowtree/vocable/presets/CategoriesFragment.kt:59]: this is a good change bc we don't need it and also it's already named categoryButton :3 👍🏼

File: app/src/main/java/com/willowtree/vocable/settings/AddPhraseViewModel.kt [hunk 2]
```diff
@@ -23,41 +23,6 @@ class AddPhraseViewModel : BaseViewModel() {
...
-        backgroundScope.launch {
-            presetsRepository.updatePhrase(phrase)
-            populateMySayings()
-
-            liveShowPhraseAdded.postValue(true)
-            delay(PHRASE_UPDATED_DELAY)
-            liveShowPhraseAdded.postValue(false)
-        }
-    }
-
```
Comment (CarolineLaw) [inline app/src/main/java/com/willowtree/vocable/settings/AddPhraseViewModel.kt:60]: how come this is getting removed? :D I am super curious!! :3

**Comment by CarolineLaw** [comment]
LOVE all of this :D you can respond to my comments or you can just ignore it and stuff, but uh yeah! 🤑 💸  LGTM 💰 💵

File: app/src/main/java/com/willowtree/vocable/settings/AddPhraseViewModel.kt (hunk 2 above)
Comment (willdenne) [reply app/src/main/java/com/willowtree/vocable/settings/AddPhraseViewModel.kt:60]: So this wasn't being used, and for the database refactor this required some changes.
So I went ahead and removed it here so I don't have to change it.

**Comment by willdenne** [comment]
@CarolineLaw Sounds good! Just as heads up,[ Google recently had a policy change](https://android-developers.googleblog.com/2022/04/expanding-plays-target-level-api-requirements-to-strengthen-user-security.html) where outdated apps will be removed on November 1st. This PR fixes that, so we'll need to get this out before then!

**Comment by CarolineLaw** [comment]
> @CarolineLaw Sounds good! Just as heads up,[ Google recently had a policy change](https://android-developers.googleblog.com/2022/04/expanding-plays-target-level-api-requirements-to-strengthen-user-security.html) where outdated apps will be removed on November 1st. This PR fixes that, so we'll need to get this out before then!

OMG!!!!!!!!!!!!  YES LOL this is certainly NOT an outdated app 😅 😭

**Pull request merged**